functions/param-names.js

```diff
@@ -15,7 +15,14 @@ module.exports = (targetVal, _opts, paths) => {
   }
 
   if (['path', 'query'].includes(targetVal.in) && targetVal.name !== 'api-version') {
-    if (!targetVal.name.match(/^[a-z][a-z0-9]*([A-Z][a-z0-9]+)*$/)) {
+    if (targetVal.name.match(/^[$@]/)) {
+      return [
+        {
+          message: `Parameter name "${targetVal.name}" should not begin with '$' or '@'.`,
+          path: [...path, 'name'],
+        },
+      ];
+    } if (!targetVal.name.match(/^[a-z][a-z0-9]*([A-Z][a-z0-9]+)*$/)) {
```

Comment:
We actually want this check _above_ line 17.  Where it is right now will not fire on header parameters.

Reply:
+1 will update (and also add the test file)